### PR TITLE
Feat(frontend): add icons in offers

### LINKED ISSUE #605: A sponsor can choose among different icons to add to  their offers
As a sponsor user, I want to choose among a library of -10- different icons to add to an offer I created.
- [x] Define list of icons
- [x] Choose icons to display
- [x] Design and create component to choose icon.
- [x] Apply
- [x] Test
- [x] Add title to icon component "Choose an icon" - "Elije un ícono".

image

### PR DESCRIPTION
**Pull request by steph222**
Resolves #605 

### What does this PR do?
A sponsor can choose among different icons to add to their offers

### How should this be tested?
- Make run
- Login as Sponsor 
- Add offer in Offers Management
- Check offers list

image

image

### CODE REVIEW

**Review by JustinCast — APPROVED** [review]
LGTM

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]
@steph222 getting this error

image

File: hasura/migrations/1620165951453_alter_table_public_offer_alter_column_icon/up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE "public"."offer" ALTER COLUMN "icon" DROP NOT NULL;
```
Comment (JustinCast) [inline hasura/migrations/1620165951453_alter_table_public_offer_alter_column_icon/up.sql:1]: Too much migrations for such a small change

File: webapp/src/routes/OffersManagement/GenericOfferFormComponent.js [hunk 2]
```diff
@@ -28,7 +28,22 @@ import Divider from '@material-ui/core/Divider'
 import { useTranslation } from 'react-i18next'
 import '@brainhubeu/react-carousel/lib/style.css'
 import DateRangePicker from '@wojtekmaj/react-daterange-picker'
-import NumberFormat from 'react-number-format';
+import NumberFormat from 'react-number-format'
+
```
Comment (JustinCast) [inline webapp/src/routes/OffersManagement/GenericOfferFormComponent.js:32]: Delete this blank space

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]
Still getting error
image

**Comment by ldrojas** [comment]
Captura de Pantalla 2021-05-07 a la(s) 15 22 20

**Comment by steph222** [comment]
> @steph222 Nice work, just one more thing, in home dir, in the offers section, one already created offer don't show any offer icon
> 
> image

This happens because the offer already existed in the database, so it will not show any icon until the offer is edited.
This is applied in this way so that the program has no problems

**Review by JustinCast — APPROVED** [review]

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]
@steph222 still getting the non icon for an offer that already existed in the database

https://user-images.githubusercontent.com/28828796/117810243-5d5aeb00-b21c-11eb-8a07-8cbf57673121.png

**Comment by steph222** [comment]
> still getting the non icon for an offer that already existed in the database

There is already a default
image

File: hasura/migrations/1620148584870_alter_table_public_offer_add_column_icon/up.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE "public"."offer" ADD COLUMN "icon" varchar NULL;
```
Comment (leisterfrancisco) [inline hasura/migrations/1620148584870_alter_table_public_offer_add_column_icon/up.sql:1]: Please, set default value and remove null

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Comment by Edgar097** [comment]
Still getting the error if I dont set up the permissions, you should merge the changes too and check the conflicts.

**Comment by steph222** [comment]
> Still getting the error if I dont set up the permissions, you should merge the changes with the master too and check the conflicts.

Thanks!

**Review by leisterfrancisco — APPROVED** [review]

**Pull request merged**